bravado/async_http_client.py

```diff
@@ -44,116 +33,26 @@ def start_request(self, request_params):
 
         :return: crochet EventualResult
         """
-        # request_params has mandatory: method, url, params, headers
+        url = '%s?%s' % (request_params['url'], urllib_utf8.urlencode(
+            request_params.get('params', []), True))
+
         request_params = {
             'method': str(request_params.get('method', 'GET')),
-            'bodyProducer': stringify_body(request_params),
-            'headers': listify_headers(request_params.get('headers', {})),
-            'uri': '%s?%s' % (
-                request_params['url'],
-                urllib_utf8.urlencode(request_params.get('params', []), True))
+            'body': stringify_body(request_params),
+            'headers': request_params.get('headers', {}),
         }
 
         # crochet only supports bytes for the url
-        if isinstance(request_params['uri'], unicode):
-            request_params['uri'] = request_params['uri'].encode('utf-8')
-
-        crochet.setup()
-        return self.fetch_deferred(request_params)
-
-    @crochet.run_in_reactor
-    def fetch_deferred(self, request_params):
-        """The main core to start the reacter and run the API
-        in the background. Also the callbacks are registered here
-
-        :return: crochet EventualResult
-        """
-        finished_resp = Deferred()
-        agent = Agent(reactor)
-        deferred = agent.request(**request_params)
-
-        def response_callback(response):
-            """Callback for response received from server, even 4XX, 5XX possible
-            response param stores the headers and status code.
-            It needs a callback method to be registered to store the response
-            body which is provided using deliverBody
-            """
-            response.deliverBody(_HTTPBodyFetcher(request_params,
-                                                  response, finished_resp))
-        deferred.addCallback(response_callback)
-
-        def response_errback(reason):
-            """Error callback method like server not reachable or conn. refused
+        if isinstance(url, unicode):
+            url = url.encode('utf-8')
```

Comment:
Nope, in python2.x if the variables used in the substitution are unicode, the final result will be unicode as well:

```python

>>> '%s' % (u'asdf')
u'asdf'
```